### PR TITLE
DDF-3142 Skipping unit tests if symlinks cannot be created

### PR DESCRIPTION
**Pull request by brjeter**
#### What does this PR do?
Fixes the build
#### Who is reviewing it? 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
@paouelle @mcalcote 
#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@figliold 
@stustison
#### How should this be tested? (List steps with links to updated documentation)
Run the unit tests
#### What are the relevant tickets?
[DDF-3142](https://codice.atlassian.net/browse/DDF-3142)
#### Checklist:
- [] Documentation Updated
- [X] Update / Add Unit Tests
- [] Update / Add Integration Tests
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/migration/ExportMigrationEntryImplTest.java [hunk 1]
```diff
@@ -286,6 +286,11 @@ public void testStoreWhenRequiredAndFileIsAbsoluteOutsideDDFHome() throws Except
 
     @Test
     public void testStoreWhenRequiredAndFileIsASoftLink() throws Exception {
+        //Cannot create symlinks reliably on Windows
+        if (isRunningOnWindows()) {
```
Comment (paouelle) [inline platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/migration/ExportMigrationEntryImplTest.java]: ✏️ Preferrable to use org.junit.Assume instead.

File: platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/migration/ExportMigrationReportImplTest.java [hunk 2]
```diff
@@ -287,6 +287,11 @@ public void testRecordExternalFileWhenUnableToComputeChecksum() throws Exception
 
     @Test
     public void testRecordExternalSoftlink() throws Exception {
+        //Cannot create symlinks reliably on Windows
+        if (isRunningOnWindows()) {
```
Comment (paouelle) [inline platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/migration/ExportMigrationReportImplTest.java]: ✏️ Preferable to use org.junit.Assume instead.

File: platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/migration/ImportMigrationEntryImplTest.java [hunk 3]
```diff
@@ -355,6 +355,11 @@ public void isNotMigratableWithAbsolutePath() {
 
     @Test
     public void isNotMigratableWithSymbolicLink() throws Exception {
+        //Cannot create symlinks reliably on Windows
+        if (isRunningOnWindows()) {
```
Comment (paouelle) [inline platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/migration/ImportMigrationEntryImplTest.java]: ✏️ Preferable to use org.junit.Assume instead.

File: platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/migration/ImportMigrationExternalEntryImplTest.java [hunk 4]
```diff
@@ -163,6 +163,11 @@ public void restoreFailsWhenFileIsNotSoftLink() throws Exception {
 
     @Test
     public void restoreRecordsWarningWhenFileIsNotNormal() throws Exception {
+        //Cannot create symlinks reliably on Windows
+        if (isRunningOnWindows()) {
```
Comment (paouelle) [inline platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/migration/ImportMigrationExternalEntryImplTest.java]: ✏️ Preferable to use org.junit.Assume instead.

File: platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/migration/PathUtilsTest.java [hunk 5]
```diff
@@ -145,6 +145,11 @@ public void testGetChecksumForWithIOException() throws Exception {
 
     @Test
     public void testGetChecksumForWithSoftlink() throws Exception {
+        //Cannot create symlinks reliably on Windows
```
Comment (paouelle) [inline platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/migration/PathUtilsTest.java]: ✏️ Preferable to use org.junit.Assume instead.

File: platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/migration/AbstractMigrationTest.java [hunk 6]
```diff
@@ -373,6 +373,11 @@ public void verifyReportHasMatchingWarning(MigrationReport report, String messag
         assertThat(warning.getMessage(), containsString(message));
     }
 
+    public boolean isRunningOnWindows() {
```
Comment (paouelle) [inline platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/migration/AbstractMigrationTest.java]: ✏️ Can be replaced with org.apache.commons.lang.SystemUtils.IS_OS_WINDOWS

**Comment by paouelle** [comment]
@eric, I like that and if it fails, simply throw back AssumptionViolatedException which will just mark it skipped instead of successful. Might even be doable in the base class method createSoftLink.

**Comment by clockard** [comment]
build now

**Pull request merged**